common/src/main/java/io/druid/common/config/Log4jShutdown.java

```diff
@@ -138,4 +140,55 @@ public boolean isStopped()
   {
     return State.STOPPED.equals(getState());
   }
+
+  private static class StateHolder
+  {
+    @GuardedBy("this")
+    private State current;
+
+    private StateHolder(State initial) {
+      current = initial;
+    }
+
+    private synchronized boolean compareAndSet(State expected, State transition)
+    {
+      if (current == expected) {
+        return transition(transition);
+      }
+      return false;
+    }
+
+    private synchronized boolean waitTransition(State expected, State transition, long timeout)
+    {
+      if (current == expected) {
+        long remaining = timeout;
+        try {
+          long prev = System.currentTimeMillis();
+          while (current != transition && remaining > 0) {
+            wait(remaining);
+            long now = System.currentTimeMillis();
+            remaining -= now - prev;
+            prev = now;
+          }
+        }
+        catch (InterruptedException e) {
+          Thread.currentThread().interrupt();
+          // ignore
```

Comment:
This should probably not be ignored. Even scala counts "InterruptedException" among its fatal exceptions (see http://www.scala-lang.org/api/2.10.2/index.html#scala.util.control.NonFatal$ for example)